controllers/operandrequest/reconcile_operand.go

```diff
@@ -948,16 +948,48 @@ func (r *Reconciler) createK8sResource(ctx context.Context, k8sResTemplate unstr
 	name := k8sResTemplate.GetName()
 	namespace := k8sResTemplate.GetNamespace()
 
+	if kind == "Route" {
+		host, _, err := unstructured.NestedString(k8sResTemplate.Object, "spec", "host")
+		klog.Info("11111 host %v: ", host)
+
+		if err != nil {
+			klog.Errorf("failed to get spec.host for Route: %v", err)
+			return err
+		}
+
+		// Calculate and set hash for spec.host
+		hostHash := util.CalculateHash(host)
+		if newAnnotations == nil {
+			newAnnotations = make(map[string]string)
+		}
+		newAnnotations["openshift.io/keycloak-host.hashedData"] = hostHash
+		// klog.Infof("22222 Set host hash annotation: %s", hostHash)
+	}
+
```

Comment:
I am wondering that if this is redundant?

It seems that variable `k8sResTemplate` only contains kind, name and namespace, and it does not have any `.spec` field yet.